opentelemetry-jaeger/src/exporter/mod.rs

```diff
@@ -341,30 +342,56 @@ impl PipelineBuilder {
 
     /// Build a configured `sdk::trace::TracerProvider` with a batch span processor using the
     /// specified runtime.
-    pub fn build_batch<R: TraceRuntime>(
+    pub fn build_batch<R: JaegerTraceRuntime>(
         mut self,
         runtime: R,
     ) -> Result<sdk::trace::TracerProvider, TraceError> {
         let (config, process) = self.build_config_and_process();
-        let exporter = self.init_exporter_with_process(process)?;
+        let exporter = self.init_exporter_with_process(process, runtime.clone())?;
         let mut builder =
             sdk::trace::TracerProvider::builder().with_batch_exporter(exporter, runtime);
         builder = builder.with_config(config);
 
         Ok(builder.build())
     }
 
+    /// Initialize a new simple exporter.
+    ///
+    /// This is useful if you are manually constructing a pipeline.
+    pub fn init_simple_exporter(mut self) -> Result<Exporter, TraceError> {
+        let (_, process) = self.build_config_and_process();
+        self.init_simple_exporter_with_process(process)
+    }
+
+    fn init_simple_exporter_with_process(self, process: Process) -> Result<Exporter, TraceError> {
+        let export_instrumentation_lib = self.export_instrument_library;
+        let uploader = self.init_simple_uploader()?;
+
+        Ok(Exporter {
+            process: process.into(),
+            export_instrumentation_lib,
+            uploader,
+        })
+    }
+
     /// Initialize a new exporter.
     ///
     /// This is useful if you are manually constructing a pipeline.
-    pub fn init_exporter(mut self) -> Result<Exporter, TraceError> {
+    pub fn init_exporter<R: JaegerTraceRuntime>(
```

Comment:
Maybe `init_batch_exporter` is a better name here? Just so it distinguishes itself from the simple exporter. Same with `init_uploader`, maybe we should use `init_batch_uploader`?

Reply:
I agree the naming (and documentation) is probably not good enough.
In practice, the simple exporter/uploader is also a batch exporter/uploader. The only difference is that it makes blocking calls to emit these batches over UDP. I didn't like calling them blocking exporter/uploader so I settled for simple instead.

Let me know what you think. I can rename the asynchronous exporter/uploader init functions `init_batch_exporter ` and `init_batch_uploader` if you think that's ok.

Reply:
Right. Maybe `SyncUploader` and `AsyncUploader` would be more suitable for this case? Similarly, we can have `init_async_exporter` and `init_sync_exporter`.